src/ts/ssh/services/keyExchangeService.ts

```diff
@@ -262,46 +276,50 @@ export class KeyExchangeService extends SshService {
 
 			// Check if the negotiated algorithm is the one preferred by THIS side.
 			// This means if there was a "guess" at kex initialization then it was correct.
-			const negotiatedKexAlgorthmIsPreferred =
+			const negotiatedKexAlgorithmIsPreferred =
 				this.exchangeContext.keyExchange === config.keyExchangeAlgorithms[0]?.name;
 
 			// If a guess was not sent, or the guess was wrong, send the init message now.
-			if (!alreadySentGuess || !negotiatedKexAlgorthmIsPreferred) {
+			if (!alreadySentGuess || !negotiatedKexAlgorithmIsPreferred) {
 				const kexAlgorithm = config.getKeyExchangeAlgorithm(this.exchangeContext.keyExchange!)!;
 				this.exchangeContext.exchange = kexAlgorithm.createKeyExchange();
-				this.exchangeContext.exchangeValue = await this.exchangeContext.exchange.startKeyExchange();
+				this.exchangeContext.exchangeValue =
+					await this.exchangeContext.exchange.startKeyExchange();
 
 				const reply = new KeyExchangeDhInitMessage();
 				reply.e = this.exchangeContext.exchangeValue;
 				await this.session.sendMessage(reply, cancellation);
+			} else {
+				this.trace(
+					TraceLevel.Verbose,
+					SshTraceEventIds.algorithmNegotiation,
+					'Already sent correct guess for key-exchange init.',
+				);
 			}
 
-			extensionInfoSignal = serverExtensionInfoSignal;
+			this.exchangeContext.isExtensionInfoRequested =
+				this.isInitialExchange &&
+				message.keyExchangeAlgorithms?.includes(serverExtensionInfoSignal);
```

Comment:
Should we add `== true` here because `message.keyExchangeAlgorithms` could be null? For all intents and purposes, I'm guessing null is treated the same as false, but it might be good to make it explicitly false.

Reply:
Conventional style in JS/TS is to rely on null/undefined being _falsy_ in boolean expressions, without being explicit about it.